common/keygen.cpp

```diff
@@ -0,0 +1,98 @@
+#include <openssl/pem.h>
+#include <memory>
+#include <random>
+#include "keygen.h"
+
+namespace {
+
+	const int kKeySize = 4096;
+
+	std::string getPubKey(RSA* rsa) {
+		
+		std::unique_ptr<BIO, decltype(&BIO_free)> buf(BIO_new(BIO_s_mem()), BIO_free);
+		PEM_write_bio_RSAPublicKey(buf.get(), rsa);
+
+		size_t length = BIO_ctrl_pending(buf.get());
+
+		std::string res;
+		res.resize(length);
+
+		BIO_read(buf.get(), res.data(), length);
```

Comment:
```suggestion
		BIO_read(buf.get(), res.data(), res.size());
```
Хотя res.size() == length, но более правильно использовать size() или length() методы.
Это убережёт от возможных ошибок и сделает код более понятны.
Мы передаём данные (res.data()) и их размер (res.size()) а не какое-то значение, не понятно как связанное с нашим объектом

Reply:
Так же неплохо бы проверить, что res.size() == BIO_read 
Т.е. что всё прочитал